Document Configuration attribute behavior in upgrade guide (#26109)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

If you build Gradle plugins written in Kotlin with this version of Gradle and need to support Gradle <7.0 you need to stick to using the Kotlin Gradle Plugin <1.9.0 and configure the Kotlin language and API levels to 1.3.
See the <<compatibility.adoc#compatibility, Compatibility Matrix>> for details about other versions.

==== Eager evaluation of `Configuration` attributes

Gradle 8.3 updates the `org.gradle.libraryelements` and `org.gradle.jvm.version` attributes of JVM Configurations
to be present at the time of creation, as opposed to previously, where they were only present after the Configuration

> had been resolved or consumed. In particular, the value for `org.gradle.jvm.version` relies on the project's configured

Reviewer: ```suggestion
had been resolved or consumed. 
In particular, the value for `org.gradle.jvm.version` relies on the project's configured
```

Reviewer: Can we put each sentence on a new line?

toolchain, meaning that querying the value for this attribute will finalize the value of the project's Java toolchain.

Plugins or build logic that eagerly queries the attributes of JVM configurations may now cause the project's Java toolchain

> to be finalized earlier than before. Attempting to modify the toolchain after it has been finalized will result in error

Reviewer: Same here and onwards.

messages similar to the following:
```
The value for property 'implementation' is final and cannot be changed any further.
The value for property 'languageVersion' is final and cannot be changed any further.
The value for property 'vendor' is final and cannot be changed any further.
```

> This sort of situation may arise when plugins or build logic eagerly queries an existing JVM Configuration's attributes

Reviewer: ```suggestion
This situation may arise when plugins or build logic eagerly queries an existing JVM Configuration's attributes to create a new Configuration with the same attributes.
```

> in order to create a new Configuration with the same attributes. Previously, this logic would have omitted the two above

Reviewer: ```suggestion
Previously, this logic would have omitted the two above
```

noted attributes entirely, while now the same logic will copy the attributes, but will also finalize the project's Java

> toolchain. In order to avoid early toolchain finalization, attribute-copying logic should be updated to lazily query

Reviewer: ```suggestion
toolchain. To avoid early toolchain finalization, attribute-copying logic should be updated to query
```

> the source Configuration's attributes:

Reviewer: ```suggestion
the source Configuration's attributes lazily:
```
